### PR TITLE
Refactor extrapolateData

### PR DESCRIPTION
**Pull request by BenjaminRueth**
* `storeData()` is now always called in `BaseCouplingScheme::accelerate()`
* update corresponding tests for extrapolation
* removes storing of data from `BaseCouplingScheme::extrapolateData(DataMap &data)`

Related to #133

### CODE REVIEW

File: src/cplscheme/BaseCouplingScheme.cpp [hunk 1]
```diff
@@ -265,7 +265,6 @@ void BaseCouplingScheme::extrapolateData(DataMap &data)
       PRECICE_DEBUG("Extrapolate data: " << pair.first);
       PRECICE_ASSERT(pair.second->oldValues.cols() > 1);
       Eigen::VectorXd &values       = pair.second->values();
-      pair.second->oldValues.col(0) = values;  // = x^t
       values *= 2.0;                           // = 2*x^t
       values -= pair.second->oldValues.col(1); // = 2*x^t - x^(t-1)
       utils::shiftSetFirst(pair.second->oldValues, values);
```
Comment (uekerman) [inline src/cplscheme/BaseCouplingScheme.cpp:270]: In principle you also store data in `shiftSetFirst`, right? Question is whether you want to change this in this PR or later?

**Review by uekerman — COMMENTED** [review]
It's a bit bad practice to change a function from private to public just to allow usage in tests. Should not be overly complicated to work with friend declarations here, right? (compare how `extrapolateData` is called)

File: src/cplscheme/BaseCouplingScheme.cpp (hunk 1 above)
Comment (BenjaminRueth) [reply src/cplscheme/BaseCouplingScheme.cpp:270]: I would prefer changing this later. https://github.com/BenjaminRueth/precice/tree/package_time overhauls this part of the extrapolation. I think changing it in this PR is not really necessary.

**Comment by BenjaminRueth** [comment]
> It's a bit bad practice to change a function from private to public just to allow usage in tests. Should not be overly complicated to work with friend declarations here, right? (compare how `extrapolateData` is called)

Encapsulation should be fixed now. See d3036ae and 8238bea.

**Review by uekerman — APPROVED** [review]

**Pull request merged**